app/test/testmerginapi.cpp

```diff
@@ -11,96 +11,106 @@
-    initTestProject();
-    copyTestProject();
+  QString apiRoot = ::getenv( "TEST_MERGIN_URL" );
+  QString username = ::getenv( "TEST_API_USERNAME" );
+  QString password = ::getenv( "TEST_API_PASSWORD" );
 
-    qDebug() << "TestMerginApi::initTestCase DONE";
-  }
+  // let's make sure we do not mess with the public instance
+  QVERIFY( apiRoot != MerginApi::defaultApiRoot() );
+
+  mApi->setApiRoot( apiRoot );
+  QSignalSpy spy( mApi, &MerginApi::authChanged );
+  mApi->authorize( username, password );
+  Q_ASSERT( spy.wait( LONG_REPLY ) );
+  QCOMPARE( spy.count(), 1 );
+
+  mUsername = username;  // keep for later
+
+  // figure out the directory with our test data (in this git repo)
+  QDir dir = QFileInfo( __FILE__ ).dir();
+  QVERIFY( dir.cd( "../../test/test_data" ) );
+  mTestDataPath = dir.absolutePath();
+
+  initTestProject();
+
+  qDebug() << "TestMerginApi::initTestCase DONE";
 }
 
-void TestMerginApi::testListProject()
+void TestMerginApi::cleanupTestCase()
 {
-  qDebug() << "TestMerginApi::testListProjectFinished START";
-  runTests++;
+  // remove test projects on the server
+  deleteRemoteProject( mUsername, TestMerginApi::TEST_PROJECT_NAME );
+  deleteRemoteProject( mUsername, TestMerginApi::TEST_PROJECT_NAME_DOWNLOAD );
+
+  // remove project data locally
+  QDir testDir( mApi->projectsPath() );
+  testDir.removeRecursively();
+  qDebug() << "TestMerginApi::cleanupTestCase DONE";
+}
 
-  QSignalSpy spy( mApi, SIGNAL( listProjectsFinished( ProjectList ) ) );
+void TestMerginApi::testListProject()
+{
+  // check that there's no hello world project
+  QSignalSpy spy0( mApi, &MerginApi::listProjectsFinished );
   mApi->listProjects( QString() );
+  QVERIFY( spy0.wait( SHORT_REPLY ) );
+  QCOMPARE( spy0.count(), 1 );
+  QVERIFY( !_findProjectByName( mUsername, "hello world project", mMerginProjectModel->projects() ) );
+
+  // create the project on the server (the content is not important)
+  createRemoteProject( mUsername, "hello world project", mTestDataPath + "/" + TEST_PROJECT_NAME + "/" );
 
+  // check the project exists on the server
+  QSignalSpy spy( mApi, &MerginApi::listProjectsFinished );
+  mApi->listProjects( QString() );
   QVERIFY( spy.wait( SHORT_REPLY ) );
   QCOMPARE( spy.count(), 1 );
+  QVERIFY( _findProjectByName( mUsername, "hello world project", mMerginProjectModel->projects() ) );
 
-  ProjectList projects = mMerginProjectModel->projects();
-  QVERIFY( !mMerginProjectModel->projects().isEmpty() );
-  qDebug() << "TestMerginApi::testListProjectFinished PASSED";
-  passedTests++;
+  // get rid of the project again
+  deleteRemoteProject( mUsername, "hello world project" );
 }
 
 /**
  * Download project from a scratch using fetch endpoint.
  */
 void TestMerginApi::testDownloadProject()
 {
-  qDebug() << "TestMerginApi::testDownloadProject START";
-  runTests++;
+  // create the project on the server (the content is not important)
+  QString projectName = "mobile_demo_mod";
```

Comment:
The name is not relevant, but anyway I would rename it to value of TEST_PROJECT_NAME.